src/Common/src/System/Diagnostics/Debug.cs

```diff
@@ -84,19 +144,33 @@ public static void WriteLine(string message)
         [System.Diagnostics.Conditional("DEBUG")]
         public static void Write(string message)
         {
-            s_logger.WriteCore(message ?? string.Empty);
+            if (message == null)
+            {
+                WriteCore(string.Empty);
+                return;
+            }
+            if (s_needIndent)
+            {
+                message = GetIndentString() + message;
+                s_needIndent = false;
+            }
+            WriteCore(message);
+            if (message.EndsWith(Environment.NewLine))
+            {
+                s_needIndent = true;
```

Comment:
I don't think we should go out of our way to make this entire thing thread safe but we should do some minimal effort like making it ThreadStatic similar to desktop.